File: app/src/main/java/com/feresr/imdino/NeuronRow.java
package com.feresr.imdino;

import java.util.ArrayList;

/**
 * Created by feresr on 23/10/2016.
 */

public class NeuronRow {

    private ArrayList<Neuron> neurons = new ArrayList<>();

    public void addNeuron(Neuron neuron) {
        neurons.add(neuron);
    }

    public float[] process(float[] inputs) {
        float[] result = new float[neurons.size()];
        for (int i = 0; i < neurons.size(); i++) {
            result[i] = neurons.get(i).process(inputs);
        }

        return result;
    }

    public int getNumberOfSynapses() {
        return neurons.get(0).getNumberOfSynapses() * neurons.size();
    }
}

File: app/src/main/java/com/feresr/imdino/NeuronRow.kt
package com.feresr.imdino

import java.util.ArrayList

/**
 * Created by feresr on 23/10/2016.
 */

class NeuronRow {

    private val neurons = ArrayList<Neuron>()

    fun addNeuron(neuron: Neuron) {
        neurons.add(neuron)
    }

    fun process(inputs: FloatArray): FloatArray {
        val result = FloatArray(neurons.size)
        for (i in neurons.indices) {
            result[i] = neurons[i].process(inputs)
        }

        return result
    }

    val numberOfSynapses: Int
        get() = neurons[0].numberOfSynapses * neurons.size
}
